javascript/javascript1/week4/class-exercises/character-frequencies.js

```diff
@@ -0,0 +1,80 @@
+function countCharacters(charactersArray) {
+  const characterCounter = {};
+  for (let i = 0; i < charactersArray.length; i++) {
+    const character = charactersArray[i];
+    if (characterCounter.hasOwnProperty(character)) {
+      characterCounter[character] += 1; //already exist
+    } else {
+      characterCounter[character] = 1; //new one
+    }
+  }
+  return characterCounter; //{ a: 1, h: 1, p: 2, y: 1 }
+}
+
+function formCharacterAndCountArray(characterCounter) {
+  const charactersAndCountArray = [];
+  for (const character in characterCounter) {
+    charactersAndCountArray.push({
+      character,
+      count: characterCounter[character],
+    });
+  }
+  return charactersAndCountArray; //[{character: 'a', count: 1},.. ]
+}
```

Comment:
You can set the properties for each letter dynamically by checking first if a record exists. If it's the first time the character is added to the object, than `isNaN(obj[letter]) will return true and the `1` will be set. If it's an already existing letter, it will simply do the + 1.

```
  for(let i = 0; i < strLength; i++) {
      const letter = str[i];
      obj[letter] = (isNaN(obj[letter]) ? 1 : obj[letter] + 1);
  }
```

It's a bit more simple solution, but honestly it wasn't the first approach that came to my mind.

Reply:
I really like it, thank you!